src/librustc_mir_build/build/matches/mod.rs

```diff
@@ -297,7 +291,7 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         guard: Option<(&Guard<'tcx>, region::Scope)>,
         fake_borrow_temps: &Vec<(Place<'tcx>, Local)>,
         scrutinee_span: Span,
-        arm_scope: region::Scope,
+        arm_scope: Option<region::Scope>,
```

Comment:
it would probably be good to update the doc-comment for this method to make it clear that it now can be applied to patterns in `match`-arms or in `let`-statements. I'm assuming there is an intended invariant here that `arm_scope` is `Some` iff we're in a `match`-arm.